cookbook/testing/integration-test-scrolling.md

```diff
@@ -0,0 +1,208 @@
+
+  @override
+  Widget build(BuildContext context) {
+    final title = 'Long List';
+
+    return MaterialApp(
+      title: title,
+      home: Scaffold(
+        appBar: AppBar(
+          title: Text(title),
+        ),
+        body: ListView.builder(
+          // Add a key to the ListView. This allows us to find the list and
+          // scroll through it in our tests
+          key: Key('long_list'),
+          itemCount: items.length,
+          itemBuilder: (context, index) {
+            return ListTile(
+              title: Text(
+                '${items[index]}',
+                // Add a key to the Text Widget for each item. This allows
+                // us to look for a particular item in the list and verify the
+                // text is correct
+                key: Key('item_${index}_text'),
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+```
+
+### 2. Instrument the app
+
+Next, we'll need to create an instrumented version of our app. This code lives
+in a file called `test_driver/app.dart`.
+
+<!-- skip -->
+```dart
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:scrollable_app/main.dart' as app;
+
+void main() {
+  // This line enables the extension
+  enableFlutterDriverExtension();
+
+  // Call the `main()` function of your app or call `runApp` with any widget you 
+  // are interested in testing.
+  app.main();
+}
+```
+
+### 3. Write a test that scrolls through the list
+
+Now, we can write our test! In this example, we need to scroll through the list
+of items and verify that a particular item exists in the list. The
+[`FlutterDriver`](https://docs.flutter.io/flutter/flutter_driver/FlutterDriver-class.html)
+class provides three methods for scrolling through lists:
+
+  - The
+  [`scroll`](https://docs.flutter.io/flutter/flutter_driver/FlutterDriver/scroll.html)
+  method allows us to scroll through a specific list by a given amount. 
+  - The
+  [`scrollIntoView`](https://docs.flutter.io/flutter/flutter_driver/FlutterDriver/scrollIntoView.html)
+  method finds a specific Widget that's currently displayed on screen and will
+  scroll it completely into view.
+  - The
+  [`scrollUntilVisible`](https://docs.flutter.io/flutter/flutter_driver/FlutterDriver/scrollUntilVisible.html)
+  method scrolls through a list until a specific Widget is visible.
+
+While all three methods work for specific use-cases, `scrollUntilVisible` is
+oftentimes the most robust option. Why?
+
+  1. If we use the `scroll` method alone, we might incorrectly assume the height
+  of each item in the list. This could lead to scrolling too much or too little.
+  2. If we use the `scrollIntoView` method, we're assuming the Widget is visible
+  on screen. In order to verify our apps work on a broad range of devices, we
+  might run our integration tests against devices with different screen sizes.
+  Whether or not a particular Widget is visible can depend on the size of the
+  screen.
```

Comment:
I like this recommendation, but as explained above "visible on screen" may not be accurate enough.